common/namespace/registry.go

```diff
@@ -199,7 +205,10 @@ func (r *registry) Stop() {
 
 func (r *registry) getAllNamespace() map[string]*Namespace {
 	result := make(map[string]*Namespace)
-	ite := r.cacheByID.Load().(cache.Cache).Iterator()
+	r.cacheLock.RLock()
+	defer r.cacheLock.RUnlock()
+
+	ite := r.cacheByID.Iterator()
 	defer ite.Close()
 
 	for ite.HasNext() {
```

Comment:
[this comment is actually for lines 223-225 but github won't let me add it there]

This comment seems wrong: both callbacks are called outside the (new) namespace cache lock, but inside the callback lock. Also the parameter names are wrong. Could you update it?

Reply:
Yep, you're right. I'll correct it.